alf/layers.py

```diff
@@ -3612,3 +3612,34 @@ def forward(self, input):
                 lambda x: x.float() if x.dtype.is_floating_point else x, input)
         with torch.cuda.amp.autocast(self._enabled):
             return self._net(input)
+
+
+class MultiHeadAttention(nn.Module):
+    """Multi-Head Attention Module."""
+
+    def __init__(self):
+        super().__init__()
+
+    def forward(self, query, key, value):
+        """Multi-Head attention computation based on the inputs.
+        Args:
+            query (Q): shape [B, head, 1, d]
```

Comment:
Can it be more generally as [B, head, M, d]?

Reply:
Good point. It can handle this more general case. Updated the comments.